src/Sprite.ts

```diff
@@ -521,26 +616,43 @@ export class Sprite extends SpriteBase {
     );
   }
 
-  *glide(seconds, x, y) {
-    const interpolate = (a, b, t) => a + (b - a) * t;
+  public *glide(seconds: number, x: number, y: number): Yielding<void> {
+    const interpolate = (a: number, b: number, t: number): number =>
+      a + (b - a) * t;
 
     const startTime = new Date();
     const startX = this._x;
     const startY = this._y;
 
     let t;
     do {
-      t = (new Date() - startTime) / (seconds * 1000);
+      t = (new Date().getTime() - startTime.getTime()) / (seconds * 1000);
       this.goto(interpolate(startX, x, t), interpolate(startY, y, t));
       yield;
     } while (t < 1);
   }
 
-  get penDown() {
+  public moveAhead(value = Infinity): void {
+    if (typeof value === "number") {
+      this._project.changeSpriteLayer(this, value);
+    } else {
+      this._project.changeSpriteLayer(this, 1, value);
```

Comment:
Can we make the type for `value` explicit here and below? It matches the `number` or third argument for `Project.changeSpriteLayer` which, oh god, also isn't typed, wut. (It's from `spritesAndClones[number]` which is `Sprite`.)

Reply:
Good catch! Fixed the argument type.